### ISSUE TITLE
Path resolution is not intuitive

### ISSUE THREAD

**Issue by pijusn** [wix/wix-http-testkit#34]
In path resolution, there are two factors:
- context path _coming from BaseUri_
- path _coming from method parameter_

For example:

```
context=/abc/
path=more/stuff
/abc/more/stuff

context=/abc/
path=stuff
/abc/stuff
```

There are a couple related problems / unintuitive (unexpected behaviours):

```
context=None
path=cats
/cats

context=None
path=/cats
//cats
```

Additionally, while context is supposed to be something _you don't have to think about in tests_, you actually care about it:

```
context=/my
path=cats
/mycats

context=/my
path=/cats
/my/cats

context=/my/
path=cats
/my/cats

context=/my/
path=/cats
/my//cats
```

So, looking at `input => result`, it's easy to understand how it works but it's not so intuitive when using the API.

**Comment by pijusn** [comment]
I am pretty sure behaviour described is exactly the same (or very close) as what you get from JDK's `Path` and `File`. I know it's a different protocol but introducing opinion to make API more usable is more important.

**Comment by liucijus** [comment]
I think url should be constructed explicitly. If you need a slash in your url, you have to specify it. For example, in case of
```
context=None
path=cats
```
it should result in
```
cats
```

**Comment by pijusn** [comment]
That is not a valid Path 😄  Path in URL must start with "/". That's a not really optional.

**Comment by liucijus** [comment]
My point is, that user has to specify valid path

**Comment by noam-almog** [comment]
I guess that it depends on how many people are failing with this, I can apply some fix layer that will remove redundant //. for some reason i though akka http will handle it, and apparently they don't

**Comment by pijusn** [comment]
Path `//something` is technically not the same as `/something`. I guess that's why it's not "fixed".

**Comment by liucijus** [comment]
@noam-almog can you make this explicit, please? Currently it causes a lot of pain debugging urls. It would be nice to have this resolved and released with third party deps until this Thursday.
In my opinion testkit should not fix any urls itself. 
cc @nadavwe

**Comment by nadavwe** [comment]
I totally agree. regarding urls - things should be simple and explicit, no magic behind the scenes.

**Comment by nadavwe** [comment]
@noam-almog for me the most important use case is:
`"" + "/a" -> "/a"`
all the others are nice but less important.
